lib/spack/docs/packaging_guide.rst

```diff
@@ -2695,6 +2781,43 @@ the build will fail and the install prefix will be removed.  If they
 succeed, Spack considers the build successful and keeps the prefix in
 place.
 
+^^^^^^^^^^^^^^^^
+Build-time tests
```

Comment:
This section LGTM in that it documents what is currently there, but I want to rework how tests are done.  I've been working with @BarrySmith on this.  We're working on fleshing out an API for doing short (smoke) tests and long tests on the build *and* install directories.  Some packages only support testing in the build directory, and others can test on an install directory, and I'd like to make it so that you can run a command like `spack short-test <spec>` or `spack long-test <spec>` *after* installation and have it work either way.   I'd also like to make that easy for the user...

So can we put a note on this section that the test API is expected to change rather drastically in future versions?